src/libsyntax/parse/lexer/mod.rs

```diff
@@ -1444,6 +1446,16 @@ impl<'a> StringReader<'a> {
                         self.mk_ident(&format!("'{}", lifetime_name))
                     });
 
+                    if c2.is_numeric() {
+                        // this is a recovered lifetime written `'1`, error but accept it
+                        self.err_span_(
+                            start_with_quote,
+                            self.pos,
+                            "lifetimes can't start with a number",
```

Comment:
```suggestion
                            "lifetime names cannot start with a number",
```

Reply:
Nit: My impression was that diagnostics usually use the "cannot" form (confirmed to some degree by grep).